base/cpuid.cpp

```diff
@@ -9,18 +10,19 @@
 #include <cpuid.h>
 #endif
 
+#include "glog/logging.h"
+
 namespace principia {
 namespace base {
 namespace _cpuid {
 namespace internal {
+
 namespace {
 
-struct CPUIDResult {
-  std::uint32_t eax;
-  std::uint32_t ebx;
-  std::uint32_t ecx;
-  std::uint32_t edx;
-};
+std::vector<CPUIDFeatureFlag>& CPUIDFlags() {
+  static std::vector<CPUIDFeatureFlag> result;
```

Comment:
This singleton is weird.  Why is this not a static member variable of `CPUIDFeatureFlag` with a selector?